homeassistant/components/bemfa/helper.py

```diff
@@ -0,0 +1,69 @@
+"""Support for bemfa ervice."""
+
+from __future__ import annotations
+
+from collections.abc import Mapping
+import hashlib
+import logging
+from typing import Any
+
+from .const import MSG_OFF, MSG_ON, MSG_SEPARATOR, TOPIC_PREFIX
+from .entities_config import ENTITIES_CONFIG, GENERATE, RESOLVE, SUFFIX
+
+_LOGGING = logging.getLogger(__name__)
+
+
+def generate_topic(domain: str, entity_id: str) -> str:
+    """Generate topic by hass entity id."""
+    suffix = ENTITIES_CONFIG[domain][SUFFIX]
+
+    # bemfa topic supports only alphabets and numbers, so we md5 the entity id.
+    return TOPIC_PREFIX + hashlib.md5(entity_id.encode("utf-8")).hexdigest() + suffix
```

Comment:
Note that HA entity IDs are alphanumeric + `.` + ` _`.
What happens when an entity_id is changed in Home Assistant?

Reply:
Thanks  emontnemery. When an entity id changed, this integration will not sync its state to Bemfa Service anymore.
One need to remove and re-config this integration at this moment.